orca-igor/src/main/java/com/netflix/spinnaker/orca/igor/IgorService.java

```diff
@@ -66,6 +66,12 @@ Map<String, Object> getBuild(
       @Path("master") String master,
       @Path(encode = false, value = "job") String job);
 
+  @GET("/builds/status/{buildNumber}/{master}")
+  Map<String, Object> getBuildWithJobAsQueryParam(
+      @Path("buildNumber") Integer buildNumber,
+      @Path("master") String master,
+      @Query(encodeValue = false, value = "job") String job);
```

Comment:
One concern here is on the encodeValue... if there are any kinda characters in the job name, I could see this not working UNLESS they're encoded.  Wouldn't think "job" but... it's a caution flag.